crates/runtime_crypto/src/methods/generate_key.rs

```diff
@@ -0,0 +1,201 @@
+use anyhow::{anyhow, Ok, Result};
+use lagon_runtime_v8_utils::{
+    extract_v8_string, extract_v8_uint32, extract_v8_uint8array, v8_string,
+};
+use num_traits::FromPrimitive;
+use once_cell::sync::Lazy;
+use ring::rand::SecureRandom;
+use ring::signature::EcdsaKeyPair;
+use rsa::pkcs1::EncodeRsaPrivateKey;
+use rsa::rand_core::OsRng;
+use rsa::{BigUint, RsaPrivateKey};
+
+use crate::{get_named_curve, get_sha_hash_from_v8_obj, CryptoNamedCurve, Sha};
+
+#[derive(Debug)]
+pub enum KeyGenAlgorithm {
+    Rsa {
+        modulus_length: u32,
+        public_exponent: Vec<u8>,
+    },
+    Ec {
+        named_curve: CryptoNamedCurve,
+    },
+    Aes {
+        length: u32,
+    },
+    Hmac {
+        hash: Sha,
+        length: Option<u32>,
+    },
+}
+
+pub fn get_generate_key_algorithm(
+    scope: &mut v8::HandleScope,
+    value: v8::Local<v8::Value>,
+) -> Result<KeyGenAlgorithm> {
+    if let Some(algorithm) = value.to_object(scope) {
+        let name_key = v8_string(scope, "name");
+        let name = match algorithm.get(scope, name_key.into()) {
+            Some(name) => extract_v8_string(name, scope)?,
+            None => return Err(anyhow!("Algorithm name not found")),
+        };
+
+        if name == "RSASSA-PKCS1-v1_5" || name == "RSA-PSS" || name == "RSA-OAEP" {
+            let modulus_length_key = v8_string(scope, "modulusLength");
+            let modulus_length = match algorithm.get(scope, modulus_length_key.into()) {
+                Some(modulus_length) => extract_v8_uint32(scope, modulus_length)?,
+                None => return Err(anyhow!("Algorithm modulusLength not found")),
+            };
+
+            let public_exponent_key = v8_string(scope, "publicExponent");
+            let public_exponent = match algorithm.get(scope, public_exponent_key.into()) {
+                Some(public_exponent) => extract_v8_uint8array(public_exponent)?,
+                None => return Err(anyhow!("Algorithm publicExponent not found")),
+            };
+
+            // let sha = get_sha_hash_from_v8_obj(scope, algorithm)?;
```

Comment:
Could we add the sha here? It seems required https://w3c.github.io/webcrypto/#dfn-RsaHashedKeyGenParams

Reply:
<img width="1728" alt="image" src="https://github.com/lagonapp/lagon/assets/47808321/3d930b44-1be3-47ac-b50e-d1c711de8720">
The problem is that when calling sign or verify, the parameter key_value: Vec<u8> is used, so I cannot know which hash RSA is using.

Reply:
Hum got it, we can tackle this later and always use a sha256 for RSA